var/spack/repos/builtin/packages/miopen-hip/package.py

```diff
@@ -27,20 +29,36 @@ class MiopenHip(CMakePackage):
     depends_on('bzip2', type='link')
     depends_on('sqlite', type='link')
     depends_on('half', type='build')
-    for ver in ['3.5.0', '3.7.0', '3.8.0']:
+
+    patch('0001-Add-rocm-path-and-rocm-device-lib-path-flags.patch', when='@3.9.0:')
+
+    for ver in ['3.5.0', '3.7.0', '3.8.0', '3.9.0', '3.10.0']:
         depends_on('hip@' + ver, type='build', when='@' + ver)
         depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
         depends_on('comgr@' + ver, type='link', when='@' + ver)
         depends_on('llvm-amdgpu@' + ver, type='build', when='@' + ver)
         depends_on('rocm-clang-ocl@' + ver, type='build', when='@' + ver)
         depends_on('rocblas@' + ver, type='link', when='@' + ver)
         depends_on('rocm-device-libs@' + ver, type='link', when='@' + ver)
+        if ver in ['3.9.0:']:
+            depends_on('zlib', type=('link'))
```

Comment:
```suggestion
    depends_on('zlib', type='link', when='@3.9.0:')
```
`ver` is a string and will never match `3.9.0:`